internals/search_fulltext.py

```diff
@@ -0,0 +1,216 @@
+
+from framework.basehandlers import FlaskHandler
+from internals.core_models import FeatureEntry
+
+
+# We consider all words that have three or more letters.
+# That includes underscores (_) and digits.
+WORD_RE = re.compile(r'\w\w\w+')
+
+# We ignore stop words because they return too many results to be useful.
+STOP_WORDS = frozenset((
+    'the and has was but was will are were with can this that have than then '
+    'for not now some '
+    'new add adds '
+    'http https www monorail com net org web bug bugs blink '
+    'gmail google chromium chrome github ').split())
+
+
+class FeatureWords(ndb.Model):
+  """A bag of words that occur in the fulltext of the feature."""
+
+  feature_id = ndb.IntegerProperty(required=True)
+  words = ndb.StringProperty(repeated=True)
+
+
+def get_strings(fe):
+  """Return a list of separate string values in the given feature entry."""
+  strings = []
+  strings.append(fe.creator_email)
+  strings.append(fe.updater_email)
+  strings.extend(fe.owner_emails)
+  strings.extend(fe.editor_emails)
+  strings.extend(fe.cc_emails)
+
+  strings.append(fe.name)
+  strings.append(fe.summary)
+  # TODO: category
+  strings.extend(fe.blink_components)
+  strings.extend(fe.search_tags)
+  strings.append(fe.feature_notes)
+
+  strings.append(fe.bug_url)
+  strings.append(fe.launch_bug_url)
+
+  # TODO: impl_status_Chrome
+  strings.append(fe.flag_name)
+  strings.append(fe.ongoing_constraints)
+
+  strings.append(fe.motivation)
+  strings.append(fe.devtrial_instructions)
+  strings.append(fe.activation_risks)
+  strings.append(fe.measurement)
+
+  strings.append(fe.initial_public_proposal_url)
+  strings.extend(fe.explainer_links)
+  # TODO: standard_maturity
+  strings.append(fe.spec_link)
+  strings.extend(fe.spec_mentor_emails)
+  strings.append(fe.interop_compat_risks)
+  strings.append(fe.all_platforms_descr)
+  strings.append(fe.tag_review)
+  # TODO: tag_review_status
+  strings.append(fe.non_oss_deps)
+  strings.append(fe.anticipated_spec_changes)
+
+  # TODO: ff_views
+  # TODO: safari_views
+  # TODO: web_dev_views
+  strings.append(fe.ff_views_link)
+  strings.append(fe.safari_views_link)
+  strings.append(fe.web_dev_views_link)
+  strings.append(fe.ff_views_notes)
+  strings.append(fe.safari_views_notes)
+  strings.append(fe.web_dev_views_notes)
+  strings.append(fe.other_views_notes)
+
+  strings.append(fe.security_risks)
+  # TODO: security_review_status
+  # TODO: privacy_review_status
+
+  strings.append(fe.ergonomics_risks)
+  strings.append(fe.wpt_descr)
+  strings.append(fe.webview_risks)
+
+  strings.extend(fe.devrel_emails)
+  strings.append(fe.debuggability)
+  strings.extend(fe.doc_links)
+  strings.extend(fe.sample_links)
+
+  return strings
+
+
+def parse_words(strings):
+  """Return a list of all searchable words in the given feature entry."""
+  words = set()
+  for s in strings:
+    # skip missing fields and eliminate apostrophes
+    if not s: continue
+    s = s.lower().replace("'", "")
```

Comment:
Would it be worth adding a `strip()` invocation here? Is there any chance a whitespace character gets here? (newline, etc.)

Reply:
There could certainly be some whitespace before, between, and after any words in the string.  The entire string could also be whitespace.  All of that should work out fine when we call `findall()` on the next line.  I'll add some test cases to verify that.